static/panes/diff.ts

```diff
@@ -0,0 +1,430 @@
+                    {id: DiffType.CompilerStdErr, name: 'Compiler stderr'},
+                    {id: DiffType.ExecStdOut, name: 'Execution stdout'},
+                    {id: DiffType.ExecStdErr, name: 'Execution stderr'},
+                    {id: DiffType.GNAT_ExpandedCode, name: 'GNAT Expanded Code'},
+                    {id: DiffType.GNAT_Tree, name: 'GNAT Tree Code'},
+                ],
+                items: [],
+                render: {
+                    option: (item, escape) => {
+                        return `<div>${escape(item.name)}</div>`;
+                    },
+                },
+                dropdownParent: 'body',
+                plugins: ['input_autogrow'],
+                onChange: value => {
+                    if (picker.classList.contains('lhsdifftype')) {
+                        this.lhs.difftype = parseInt(value as any as string);
+                        this.lhs.refresh();
+                    } else {
+                        this.rhs.difftype = parseInt(value as any as string);
+                        this.rhs.refresh();
+                    }
+                    this.updateState();
+                },
+            });
+
+            if (picker.classList.contains('lhsdifftype')) {
+                this.selectize.lhsdifftype = instance;
+            } else {
+                this.selectize.rhsdifftype = instance;
+            }
+        });
+
+        this.domRoot[0].querySelectorAll('.diff-picker').forEach(picker => {
+            if (!(picker instanceof HTMLSelectElement)) {
+                throw new Error('.difftype-picker is not an HTMLSelectElement');
+            }
+            const instance = new TomSelect(picker, {
+                sortField: 'name',
+                valueField: 'id',
+                labelField: 'name',
+                searchField: ['name'],
+                options: [],
+                items: [],
+                render: {
+                    option: function (item, escape) {
+                        const origin = item.editorId !== false ? 'Editor #' + item.editorId : 'Tree #' + item.treeId;
+                        return (
+                            '<div>' +
+                            `<span class="compiler">${escape(item.compiler.name)}</span>` +
+                            `<span class="options">${escape(item.options)}</span>` +
+                            '<ul class="meta">' +
+                            `<li class="editor">${escape(origin)}</li>` +
+                            `<li class="compilerId">${escape(getItemDisplayTitle(item))}</li>` +
+                            '</ul>' +
+                            '</div>'
+                        );
+                    },
+                },
+                dropdownParent: 'body',
+                plugins: ['input_autogrow'],
+                onChange: value => {
+                    if (!((value as any as string) in this.compilers)) return;
```

Comment:
Assume this is another case of value being `string | string[] | number`? I guess this is the best we can do

Reply:
Yeah it's another case of the tom select typing weirdness, I'd like to find a better general solution for this...